code/modules/reagents/reagent_dispenser.dm

```diff
@@ -142,6 +158,23 @@
 			log_bomber(user, "detonated a", src, "via welding tool")
 			boom()
 		return
+	if(istype(I, /obj/item/assembly_holder) && accepts_rig)
+		if(rig)
+			to_chat(user, "<span class='warning'>There is another device in the way.</span>")
+			return ..()
+		user.visible_message("[user] begins rigging [I] to [src].", "You begin rigging [I] to [src]")
+		if(do_after(user, 20, target = src))
+			user.visible_message("<span class='notice'>[user] rigs [I] to [src].</span>", "<span class='notice'>You rig [I] to [src].</span>")
+			var/obj/item/assembly_holder/holder = I
+			if(locate(/obj/item/assembly/igniter) in holder.assemblies)
+				rig = holder
+				if(!user.transferItemToLoc(holder, src))
+					return
+				var/icon/test = getFlatIcon(holder)
```

Comment:
Why do you need to getFlatIcon? Can you make this use vis_contents or just use `holder.appearance`?

Reply:
Also better variable name.